src/BlockTemplatesController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Utils\ThemeUtils;
+
+/**
+ * BlockTypesController class.
+ *
+ * @internal
+ */
+class BlockTemplatesController {
+
+	/**
+	 * Holds the path for the directory where the block templates will be kept.
+	 *
+	 * @var string
+	 */
+	private $templates_directory;
+
+	/**
+	 * Directory name of the block template directory.
+	 *
+	 * @var string
+	 */
+	const TEMPLATES_DIR_NAME = 'block-templates';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->templates_directory = plugin_dir_path( __FILE__ ) . '../templates/' . self::TEMPLATES_DIR_NAME;
```

Comment:
Would using `__DIR__` here make the code easier to understand? I was a bit confused by the two dots in `../templates/`.
```suggestion
		$this->templates_directory = plugin_dir_path( __DIR__ ) . 'templates/' . self::TEMPLATES_DIR_NAME;
```